examples/stream-cycle.rs

```diff
@@ -0,0 +1,25 @@
+//! Repeats given stream values and sum them
+
+#![feature(async_await)]
+
+use async_std::io;
+use async_std::prelude::*;
+use async_std::stream;
+use async_std::task;
+
+fn main() -> io::Result<()> {
+    task::block_on(async {
+        let mut s = stream::cycle(vec![6, 7, 8]);
+        let mut total = 0;
+
+        while let Some(v) = s.next().await {
+            total += v;
+            if total == 42 {
+                println!("Found {} the meaning of life!", total);
+                break;
+            }
+        }
+
+        Ok(())
+    })
+}
```

Comment:
This example is cool, but I think it's probably fine to just have the doc test for now. In #131 and #129 there are quite a few more methods, and I think it'd be wise to try and keep our line count down where possible.